nearcore/src/lib.rs

```diff
@@ -274,6 +343,21 @@ pub fn apply_store_migrations(path: &Path, near_config: &NearConfig) {
         let db_version = get_store_version(path);
         debug_assert_eq!(db_version, near_primitives::version::DB_VERSION);
     }
+
+    // DB migration was successful, remove the checkpoint to avoid it taking up precious disk space.
+    if near_config.config.use_checkpoints_for_db_migration {
+        if let Some(checkpoint_path) = checkpoint_path {
+            info!(target: "near", "Deleting DB checkpoint at '{}'", checkpoint_path.display());
+            match std::fs::remove_dir_all(&checkpoint_path) {
+                Ok(_) => {
+                    info!(target: "near", "Deleted DB checkpoint at '{}'", checkpoint_path.display());
+                }
+                Err(err) => {
+                    error!(target: "near", "Failed to delete a DB checkpoint at '{}'. Error: {:#?}. Please delete it manually before the next start of the node.", checkpoint_path.display(), err);
+                }
+            }
+        }
+    }
```

Comment:
```suggestion
        if let Some(checkpoint_path) = checkpoint_path {
            info!(target: "near", "Deleting DB checkpoint at '{}'", checkpoint_path.display());
            match std::fs::remove_dir_all(&checkpoint_path) {
                Ok(_) => {
                    info!(target: "near", "Deleted DB checkpoint at '{}'", checkpoint_path.display());
                }
                Err(err) => {
                    error!(target: "near", "Failed to delete a DB checkpoint at '{}'. Error: {:#?}. Please delete it manually before the next start of the node.", checkpoint_path.display(), err);
                }
            }
        }
```

The `Option` is a bearer of "do we have a checkpoint" info, we don't need any extra checks.